app/javascript/packages/phone-input/index.js

```diff
@@ -46,32 +45,34 @@ export class PhoneInput extends HTMLElement {
   countryCodePairs = {};
 
   connectedCallback() {
-    /** @type {HTMLInputElement?} */
-    this.textInput = this.querySelector('.phone-input__number');
-    /** @type {HTMLSelectElement?} */
-    this.codeInput = this.querySelector('.phone-input__international-code');
-    this.codeWrapper = this.querySelector('.phone-input__international-code-wrapper');
-    this.exampleText = this.querySelector('.phone-input__example');
-
-    try {
-      this.deliveryMethods = JSON.parse(this.dataset.deliveryMethods || '');
-      this.countryCodePairs = JSON.parse(this.dataset.translatedCountryCodeNames || '');
-    } catch {}
-
-    if (!this.textInput || !this.codeInput) {
-      return;
-    }
+    if (this.isConnected) {
```

Comment:
Aside from understanding whether this is truly necessary, we could have avoided indenting the whole function by implementing this as an early return instead.

```ts
if (!this.isConnected) {
  return;
}

// ...
```